### PR TITLE
test: exercise standings and pit timing against real captured telemetry

### PR DESCRIPTION
**Pull request by CtrlAltDale**
## Description

The two hooks every standings and relative overlay is built on were among the least covered code in the app — the relative assembly at **7.8%** statements, the class assembly at **30%**. Both sit behind a dashboard config, a session and several channels at once, which is why they had gone untested despite being what the widgets actually render.

This adds a way to test them against telemetry iRacing really produced, then uses it.

**Replay fixtures.** Short windows lifted from real captures by `tools/telemetry-replay/extract-capture-fixture.py`, downsampled, cut to the variables a scenario needs, floats rounded, per-car arrays trimmed to the size of the field. Rosters are anonymised: customer IDs become sequential from 900000 and names become famous racers, so no real participant's identity ships and a fixture cannot be mistaken for a genuine entry list.

Two scenarios:

- **`multiclass-road-america.json`** — 60s of a three-class race finish, 44 cars in the world, 4 on pit road, with session results and the qualifying grid.
- **`driver-swap-portimao.json`** — a team-race pit stop where the car changes drivers mid-stop.

**A harness** (`src/testing/`) that runs a fixture through the real `StandingsProcessor` and `RelativeGapProcessor` and publishes their snapshots on a stub channel bridge, so hooks see the shapes they see in the app rather than hand-built snapshots that drift from the processors.

**Coverage built on it, plus direct unit tests** for pure modules that everything leans on and nothing tested:

| | before | after |
|---|---|---|
| `useDriverPositions.tsx` | 7.8% | 84.5% |
| `useDriverStandings.tsx` | 30.4% | 89.9% |
| `createStandings.ts` | 75.5% | 88.2% |
| `relativeGapHelpers.ts` | 71.7% | 100% |
| `colors.ts` | 63.2% | 100% |
| `referenceLapInterpolation.ts` | 82.4% | 94.1% |
| `iratingGain.ts` | 91.4% | 93.1% |
| `FuelProjectionEngine.ts` | 86.9% | 89.3% |
| overall statements | 68.7% | 70.5% |
| overall branches | 59.0% | 61.5% |

The driver-swap fixture also pins #697 against the telemetry that bug was found in: with the fix reverted the stop measures 64s, with it the real 70-plus.

**Things the real data corrected**, each of which would have been wrong in a hand-written fixture and would not have failed:

- A large entry list is not a large field. The first capture tried had 45 drivers on the roster and exactly one car in the world all session.
- A car can be in the world with no position yet, so position assertions cover placed cars rather than running ones.
- `DriverCarIdx` and `PaceCarIdx` cannot be guessed from the roster. Getting the player wrong silently empties every relative window.
- A session file is overwritten as the session runs, so its results are the final classification. Paired with mid-race telemetry the standings are incoherent — cars a lap ahead of the leader, gaps shrinking down the order. The fixture is cut from the race finish so the two agree.
- Gaps only order monotonically among cars on the lead lap; once a car is laps down its gap is dominated by the laps lost.

**Note on repo weight:** this commits ~600 KB of fixture JSON. Discussed on Discord beforehand. They are already trimmed hard — scenario-specific variable subsets, downsampling, rounded floats, arrays cut to the field. The alternative is hosting them outside the repo, at the cost of tests that cannot run standalone.

## Screenshots

Not applicable — this PR adds tests and test tooling only, with no runtime or visual change.

### Before

No change to application behaviour.

### After

No change to application behaviour.

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Performance improvement
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [ ] Dependency update

Tests and test tooling only — none of the above categories fit. No production code is modified.

## Checklist

- [x] I have discussed this change in the discord server
- [ ] I have tested this in iRacing (either in an online session or with AI)
- [x] All tests pass locally via `npm test`
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run `npm run lint` and fixed any issues
- [x] I have performed a self-review of my own code
- [ ] I have added/updated Storybook stories for visual changes
- [ ] I have updated the README.md (if applicable)
- [ ] I have updated defaultDashboard.ts if introducing new widgets or configurations (if applicable)

Not tested in iRacing: this PR changes no runtime behaviour, so there is nothing to observe in a live session. The fixtures themselves come from real captured sessions.

## Validation

- `npm test` — 1899 tests across 195 files, all passing.
- `npm run lint` — clean (`tsc --noEmit` and `eslint --max-warnings 0`).
- **Mutation checked**, rather than trusted because they were green. Two rounds:
  - `project()` replaced with a function ignoring its input and returning a constant: the first draft of the fuel tests all survived, because they asserted only that a number was above zero. Rewritten to assert the value each branch produces; five now fail against that mutant.
  - `StandingsProcessor.onFrame` replaced with a no-op: three replay tests survived because they were vacuously true on empty data — an empty relative window is trivially ordered and trivially within its size bound. Guarded; the replay specs now catch that mutant 11 of 13 rather than 8.
  - The two that still survive both mutants guard the fixture rather than the processor, and are kept in their own `describe` named for that, so they are not read as processor coverage.
- No Storybook changes; no live-environment testing performed.

## Summary by CodeRabbit

- **Tests**
  - Added replay-based coverage for multiclass standings, driver positions, pit timing, class gaps, and reference-lap behavior.
  - Added edge-case tests for fuel projections, rating gains, color handling, and lap interpolation.
  - Expanded validation of qualifying, pit-road, pace-car, and relative-position scenarios.

- **Testing Tools**
  - Added utilities for replaying telemetry fixtures and mounting dashboard state during tests.
  - Added a command-line tool for extracting compact, anonymized telemetry fixtures.

### CODE REVIEW

File: tools/telemetry-replay/extract-capture-fixture.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        if not line:
+            continue
+        try:
+            frames.append(json.loads(line))
+        except json.JSONDecodeError:
+            continue
+    proc.stdout.close()
+    proc.wait()
+    unzstd.wait()
+    return frames
```
[the remark this answers is not in the document]
Comment (CtrlAltDale) [reply tools/telemetry-replay/extract-capture-fixture.py]: Fixed in 5d67b377. Both return codes are now checked and the run aborts instead of writing a short fixture, and `jq`'s stderr is surfaced in the failure message.

`jq`'s stderr goes to a temp file rather than a pipe: stdout is drained to completion first, so a stderr pipe that filled in the meantime would deadlock.

Verified against a real 38 MB capture with its shard truncated to 40%:

```
old: exit=0, wrote a fixture with 1551 frames
new: exit=1, "zstd failed (1) reading shards in /tmp/truncated-capture"
```

Worth catching — a quietly short fixture reads as a legitimate racing scenario, so nothing downstream would have flagged it.

**Review by tariknz — APPROVED** [review]
